modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductVisibility.testcase

```diff
@@ -48,12 +48,71 @@ definition {
 	}
 
 	@description = "This is a test for COMMERCE-6086. AddAChannelFilter"
-	@ignore = "true"
 	@priority = "3"
 	test AddAChannelFilter {
 		property portal.acceptance = "false";
 
-		// TODO COMMERCE-6086 AddAChannelFilter
+		CommerceNavigator.gotoPortlet(
+            category = "Product Management",
+            portlet = "Catalogs");
+        
+		Click(locator1 = "Button#PLUS");
+        
+		CommerceEntry.addNewCatalogsEntry(
+            catalogCurrency = "US Dollar",
+            catalogLanguage = "English (United States)",
+            catalogName = "Master");
+        
+		AssertClickNoError(
+            locator1 = "Button#SUBMIT",
+            value1 = "Submit");
+                
+		CommerceNavigator.gotoPortlet(
+            category = "Store Management",
+            portlet = "Channels");
+        
+		Click(locator1 = "Button#PLUS");
+        
+		CommerceEntry.addNewChannelsEntry(
+            channelCurrency = "US Dollar",
+            channelName = "Test Channel",
+            channelType = "Site",
+            siteName = "Guest");
+        
+		CommerceNavigator.gotoPortlet(
+            category = "Product Management",
+            portlet = "Products");
+        
+		CPCommerceCatalog.newProductsEntry(
+            productCatalog = "Master",
+            productName = "Simple T-Shirt",
+            productType = "Simple");
+        
+		CommerceEntry.gotoMenuTab(menuTab = "Visibility");
+        
+		Click(locator1 = "ToggleSwitch#ENABLE_SPECIFIC_TOGGLE_SWITCH");
+        
+		Click(locator1 = "Button#PLUS");
+        
+		SelectFrame.selectFrameNoLoading(
+            locator1 = "IFrame#MODAL_BODY");
+        
+		Check(
+            locator1 = "Checkbox#ENABLE_CHECKBOX_SELECT_CHANNEL",
+            key_checkboxName = "Test Channel");
```

Comment:
It's possible to use Checkbox#ENABLE_CHECKBOX with key_checkboxName = "Test Channel"